src/main/java/com/questhelper/requirements/sailing/BoatResistanceType.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2025, Zoinkwiz <https://github.com/Zoinkwiz>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.questhelper.requirements.sailing;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import net.runelite.api.Client;
+import net.runelite.api.gameval.VarbitID;
+
+@AllArgsConstructor
+@Getter
+public enum BoatResistanceType
+{
+	KELP("kelp", VarbitID.SAILING_SIDEPANEL_BOAT_TANGLEDKELP_RESISTANT, false),
+	ICE("ice", VarbitID.SAILING_SIDEPANEL_BOAT_ICYSEAS_RESISTANT, false),
+	CRYSTAL("crystal", VarbitID.SAILING_SIDEPANEL_BOAT_CRYSTALFLECKED_RESISTANT, false),
+	FETID_WATER("fetid water", VarbitID.SAILING_SIDEPANEL_BOAT_FETIDWATER_RESISTANT, false),
+	RAPID("rapid", VarbitID.SAILING_SIDEPANEL_BOAT_RAPIDRESISTANCE, true),
+	STORM("storm", VarbitID.SAILING_SIDEPANEL_BOAT_STORMRESISTANCE, true);
+
+	private final String displayName;
+	private final int varbitId;
+	private final boolean showLevel;
+
+	public int getActiveBoatLevel(Client client)
```

Comment:
`getActiveBoatLevel` returns the current boat's resistance level to the given type, while `getBoatSlotLevel` returns the boat in the given slot's resistance level to the given type right?

Reply:
That's what it should do from my thinking on it, yeah. The thinking being we can show if the player is currently on a boat which has the right resistances to do something (which can also be another player's boat), or we can indicate if a player has the requisite components on their boat for doing something prior to getting onto a boat.